watertap/unit_models/electrocoagulation.py

```diff
@@ -0,0 +1,991 @@
+            self.charge_transfer_number.set_value(2)
+            self.stoic_coeff.set_value(1)
+            self.density_electrode_material.set_value(7860)
+
+            if (
+                self.config.overpotential_calculation
+                == OverpotentialCalculation.detailed
+            ):
+                # Dubrawski et al., 2014; Zhang et al., 2020
+                self.anode_cell_potential_std.set_value(-0.41)
+                # = S / (Z * F); Bratsch, 1989; Dubrawski et al., 2014
+                self.anode_entropy_change_std.set_value(7e-5)
+                # Electrochemical Thermodynamics and Kinetics, pg 276
+                self.anodic_exchange_current_density.set_value(2.5e-4)
+                # Electrochemical Thermodynamics and Kinetics, pg 276
+                self.cathodic_exchange_current_density.set_value(1e-3)
+
+        self.floc_basin_vol = Var(
+            initialize=1,
+            bounds=(0, None),
+            units=pyunits.m**3,
+            doc="Floc basin volume total (flotation + sedimentation)",
+        )
+
+        self.floc_retention_time = Var(
+            initialize=30,
+            bounds=(2, 200),
+            units=pyunits.minute,
+            doc="Floc basin retention time",
+        )
+
+        self.coagulant_dose = Var(
+            initialize=1,
+            bounds=(0, None),
+            units=pyunits.g / pyunits.liter,
+            doc="Assumed coagulant dose to the feed in g/L",
+        )
+
+        self.electrode_thickness = Var(
+            initialize=0.001,
+            bounds=(0, 0.1),
+            units=pyunits.m,
+            doc="Electrode thickness",
+        )
+
+        self.electrode_mass = Var(
+            initialize=10,
+            bounds=(0, None),
+            units=pyunits.kg,
+            doc="Electrode mass",
+        )
+
+        self.electrode_volume = Var(
+            initialize=1,
+            bounds=(0, None),
+            units=pyunits.m**3,
+            doc="Electrode volume",
+        )
+
+        self.cathode_area = Var(
+            initialize=1,
+            bounds=(0, None),
+            units=pyunits.m**2,
+            doc="Area of cathode",
+        )
+
+        self.anode_area = Var(
+            initialize=1,
+            bounds=(1e-6, None),
+            units=pyunits.m**2,
+            doc="Area of anode",
+        )
+
+        self.electrode_gap = Var(
+            initialize=0.005,
+            bounds=(0.00001, 0.5),
+            units=pyunits.m,
+            doc="Electrode gap",
+        )
+
+        self.electrolysis_time = Var(
+            initialize=30,
+            bounds=(0.1, 200),
+            units=pyunits.minute,
+            doc="Electrolysis time",
+        )
+
+        self.cell_volume = Var(
+            initialize=1,
+            bounds=(0, None),
+            units=pyunits.m**3,
+            doc="Volume of electrolytic cell",
+        )
+
+        self.current_density = Var(
+            initialize=1,
+            bounds=(0, 2000),
```

Comment:
Should we remove the upper bound here?

Reply:
I am pretty sure this is an astronomically high current density upper bound for EC, no one should ever hit this.

Though it is true that I sort of added upper bounds randomly across this unit model.